Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/InMemorySpawnableAssetContainer.cpp

```diff
@@ -213,63 +212,91 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
-
-            if (loadBehavior == AZ::Data::AssetLoadBehavior::NoLoad)
-            {
-                continue;
-            }
+        // Get the referenced assets directly from the product. This is done for two reasons:
+        //  1. Avoids calls to the Asset Manager for assets that are already loaded.
+        //  2. Gets the exact asset to load to avoid issues with assets that don't reload.
+        AZStd::vector<AZ::Data::Asset<AZ::Data::AssetData>*> blockingAssets;
 
-            AZ::Data::AssetId assetId = asset.GetId();
-            AZ::Data::AssetType assetType = asset.GetType();
-
-            asset = AZ::Data::AssetManager::Instance().GetAsset(assetId, assetType, loadBehavior);
-
-            if (!asset.GetId().IsValid())
+        for (AZ::Data::Asset<AZ::Data::AssetData>& asset : spawnable.m_assets)
+        {
+            AZ::SerializeContext* sc = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(sc, &AZ::ComponentApplicationBus::Events::GetSerializeContext);
+            AZ_Assert(
+                sc, "Unable to locate Serialize Context while resolving asset references in the in-memory spawnable asset container.");
+
+            auto callback = [&blockingAssets](
+                                void* object, const AZ::SerializeContext::ClassData* classData,
+                                [[maybe_unused]]const AZ::SerializeContext::ClassElement* elementData) -> bool
             {
-                AZ_Error("Prefab", false, "Invalid asset found referenced in scene while entering game mode");
-                continue;
-            }
+                if (classData->m_typeId == AZ::GetAssetClassId())
+                {
+                    auto asset = reinterpret_cast<AZ::Data::Asset<AZ::Data::AssetData>*>(object);
+
+                    if (!asset->GetId().IsValid())
+                    {
+                        AZ_Error("Prefab", false, "Invalid asset found referenced in scene while entering game mode");
+                        return false;
+                    }
+
+                    if (asset->GetStatus() != AZ::Data::AssetData::AssetStatus::NotLoaded)
+                    {
+                        // Already loaded so no need to do anything.
+                        return false;
+                    }
+
+                    const AZ::Data::AssetLoadBehavior loadBehavior = asset->GetAutoLoadBehavior();
+                    if (loadBehavior == AZ::Data::AssetLoadBehavior::NoLoad)
+                    {
+                        return false;
+                    }
+
+                    AZ::Data::AssetId assetId = asset->GetId();
+                    AZ::Data::AssetType assetType = asset->GetType();
+
+                    // Always queue load as the next step will stop loading for all PreLoad assets
+                    *asset = AZ::Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::QueueLoad);
```

Comment:
I think you're mixing semantics on the load behavior here.  All you really want (I think) is just to call asset->QueueLoad().  By calling GetAsset(), you're creating a _new_ asset<T> reference, and you're initializing the loadBehavior to QueueLoad, which has nothing to do with the current load.  It just means that the asset is requesting to be Queued whenever the parent owning the asset reference is loaded.

Reply:
You're correct, so I updated it use `QueueLoad`. The focus was more on tricking a `PreLoad` asset to be async loaded through `QueueLoad`. As it turns out explicitly setting the auto load behavior to `QueueLoad` works just as well and later setting it back also makes a lot more sense in that case.